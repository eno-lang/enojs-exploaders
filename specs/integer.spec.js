const eno = require('enojs');
const loaders = require('../index.js');

describe('integer', () => {

  let document;

  describe('42', () => {
    it('returns 42', () => {
      document = eno.parse('integer: 42');
      expect(document.field('integer', loaders.integer)).toBe(42);
    });
  });

  describe('-42', () => {
    it('returns -42', () => {
      document = eno.parse('integer: -42');
      expect(document.field('integer', loaders.integer)).toBe(-42);
    });
  });

  describe('42.0', () => {
    it('throws an error', () => {
      document = eno.parse('integer: 42.0');
      expect(() => document.field('integer', loaders.integer)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('42,0', () => {
    it('throws an error', () => {
      document = eno.parse('integer: 42,0');
      expect(() => document.field('integer', loaders.integer)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('4 2', () => {
    it('throws an error', () => {
      document = eno.parse('integer: 4 2');
      expect(() => document.field('integer', loaders.integer)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('fortytwo', () => {
    it('throws an error', () => {
      document = eno.parse('integer: fortytwo');
      expect(() => document.field('integer', loaders.integer)).toThrowErrorMatchingSnapshot();
    });
  });
});
