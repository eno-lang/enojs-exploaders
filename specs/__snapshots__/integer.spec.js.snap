// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integer 4 2 throws an error 1`] = `
"'integer' must be an integer.

   Line | Content
 >    1 | integer: 4 2
"
`;

exports[`integer 42,0 throws an error 1`] = `
"'integer' must be an integer.

   Line | Content
 >    1 | integer: 42,0
"
`;

exports[`integer 42.0 throws an error 1`] = `
"'integer' must be an integer.

   Line | Content
 >    1 | integer: 42.0
"
`;

exports[`integer fortytwo throws an error 1`] = `
"'integer' must be an integer.

   Line | Content
 >    1 | integer: fortytwo
"
`;
